src/settings.js

```diff
@@ -22,6 +22,13 @@ class Settings extends SafeEventEmitter {
     const oldSettings = storage.get(SETTINGS_STORAGE_KEY);
     settings = {...defaultSettings, ...oldSettings};
 
+    if (oldSettings != null && oldSettings.version !== process.env.EXT_VER) {
+      const oldEmoteMenuValue = oldSettings[SettingIds.LEGACY_EMOTE_MENU];
+      const emoteMenuValue = oldEmoteMenuValue ? EmoteMenuTypes.LEGACY : EmoteMenuTypes.NONE;
+      settings = {...settings, [SettingIds.EMOTE_MENU]: emoteMenuValue};
```

Comment:
replacing the settings object for every migration will be slower over time. maybe there's a way to do all mutations and then change the reference?